### DISCUSSION TITLE
Using TacticalRMM to Deploy New Clients?

### DISCUSSION THREAD

**Discussion by modem-19** [amidaware/tacticalrmm#2016] [General]
This is something I thought about the other day and want to see how feasible it might be.

First off, I have a small business that has slowly evolved from general PC building/sales to IT consulting and in some cases with some clients... hybrid MSP model.  

With that said, one of the things I've ran into with most of my clients is that I handle the entire setup of their PC's from unboxing to final deployment to the end user.   I've looked at software like ImmyBot which would work wonders, but is beyond cost prohibit for me for the monthly fees.  But I love the idea.  Have one agent that pushes out pre-scripted setups for different tenants then has a centrally managed portal where additional software can be pushed out in the future.

TRMM definitely does the latter part of using Chocolaty to push out manual software updates (works great for me thus far!)  But... is it possible to create a new TRMM tenant and then use TRMM's agent upon it's install to report back and kick off a script that then installs/configures the pre-defined software for a given client?

Say Client A needs Firefox, Chrome, Brave, Adobe, and Winamp.  Client B needs Brave, Notepad++, and Media Player Classic.  I would unbox each PC for each client, install TRMMs agent downloaded for each specific client.  Boot the PC for the first time, go through the setup screen, land at the desktop, install TRMM and then TRMM kicks off the software installs automatically?

Just thinking outloud here and seeing what the possibilities are.

Thanks!!

**Comment by dinger1986** [comment]
Yes absolutely, tasks can be setup including onboarding and install of software can be part of that

**Reply by modem-19** [reply]
@dinger1986   Where is a good starting point to learn more about that?  I haven't seen much in documentation or Youtube tutorials or such on it.

**Reply by dinger1986** [reply]
Have a play with it, just automation policys and onboarding tasks is what you want to look at then assign them to customers. Something I need to do more with our customers just not got round to it

**Reply by modem-19** [reply]
I think I need to figure out how the automation policies are configured before I can dive into onboarding.

**Reply by dinger1986** [reply]
Sure, take a look here https://docs.tacticalrmm.com/functions/automation_policies/. If anything isn't clear let us know

**Reply by modem-19** [reply]
@dinger1986  Thanks for sharing that link.  I've been reading over the documentation and trying to get more familiar with the Automation policy setup.  I still have a few questions which may be to be not being familiar with the menu/work flow setup of tRMM.

When inside of Settings > Automation Manager  I click on my "Test PC Deployment" policy, Then at the bottom I click on 'New > Script Check' to select a script.  My main question is that the correct method for getting actions tied to my Automation policy?

If so, do I then need to add multiple scripts in order to perform different actions like install software, add a local admin user (or one large custom script I write for deploying all actions, etc)?

Please advise if my thinking is correct with this.  

Thank you!

**Reply by dinger1986** [reply]
Hello,

Yes that's who's you would do it but I think you want to be using tasks not checks as tasks are ran on the agent but checks run from the server. Also tasks can be ran as "onboarding" so essentially run once.

Up to you if you do one big script or have a lot of individual scripts, either method will work
